app/views/shared/_breadcrumbs.html.slim

```diff
@@ -10,7 +10,7 @@
   -if @collection.is_a?(DocumentSet) && @collection.show_to?(current_user)
     -path.push(link_to @collection.collection.title, collection_path(@collection.collection.owner, @collection.collection))  
   -path.push(link_to @collection.title, collection_path(@collection.owner, @collection))
--elsif @collection && @collection.title.present? && @collection.is_a?(DocumentSet)
+-elsif @collection && @collection.title.present? && @collection.is_a?(DocumentSet) && @collection.is_public? && !@collection.collection.restricted?
```

Comment:
Can we try using `@collection.is_public? || @collection.show_to?(current_user)` here?  There is a scenario in which a user is looking at a document set in a private collection, but they have been added as a collaborator on that private collection so they should be able to see it.

Reply:
@benwbrum The check that you have provided in feedback will not work for the required scenario. I have updated the check and added a check for checking the collaborator of collection while showing the link in breadcrumbs. I have pushed the changes.